### PR TITLE
[Artifactory HA] Support single node installations in artifactory ha chart

### PR DESCRIPTION
**Pull request by rahulsadanandan**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by danielezer — APPROVED** [review]

File: stable/artifactory-ha/templates/artifactory-node-pdb.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if gt (int .Values.artifactory.node.replicaCount) 0 }}
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-node-pdb.yaml:1]: Not sure we should do it like this. Many times, we scale the StatefulSet directly and not via the helm chart.
If no errors, please leave this PDB.

File: stable/artifactory-ha/templates/artifactory-service.yaml [hunk 2]
```diff
@@ -49,7 +49,9 @@ spec:
   {{- end }}
   {{- end }}
   selector:
-{{- if eq .Values.artifactory.service.pool "members" }}
+{{- if eq (int .Values.artifactory.node.replicaCount) 0 }}
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-service.yaml:52]: Same here. If no error, leave this for manual scaling.

Comment (rahulsadanandan) [reply stable/artifactory-ha/templates/artifactory-service.yaml:52]: As discussed , following with the same approach.

File: stable/artifactory-ha/templates/artifactory-node-pdb.yaml (hunk 1 above)
Comment (rahulsadanandan) [reply stable/artifactory-ha/templates/artifactory-node-pdb.yaml]: This is removed to the old code.

**Review by eldada — APPROVED** [review]

**Review by chukka — APPROVED** [review]

**Pull request merged**